services/spar/test-integration/Test/LoggingSpec.hs

```diff
@@ -18,3 +26,33 @@ spec = describe "logging" $ do
         Log.flush logger
       out `shouldContain` "hrgh  woaa"
       out `shouldNotContain` "hrgh\n\nwoaa"
+
+  let withApp :: (Application -> Env -> Expectation) -> TestSpar ()
+      withApp testcase = do
+        (app, env) <- liftIO . mkApp =<< asks (^. teOpts)
+        liftIO $ testcase app env
+
+      test400 :: HW.WaiSession HW.SResponse -> Int -> TestSpar ()
+      test400 badReq expectedStatus = withApp $ \app env -> do
+        (out, resp) <- capture $ do
+          resp <- HW.withApplication app $ badReq
+          Log.flush (sparCtxLogger env)
+          pure resp
+        statusCode (HW.simpleStatus resp) `shouldBe` expectedStatus
+        out `shouldNotContain` "Expected exception instead of response with status >= 400"
+
+  it "Errors are thrown as IO Error, not handled by servant (1)" $
+    test400 (HW.post "/sso/finalize-login" "") 400
```

Comment:
It's not clear to me how this test body relates to the test behaviour that's specified;

Maybe change the helper to `captureLogging` and do the assertion on the logging itself in the test body?

Reply:
I just realized that these tests don't really make sense any more.  test400 was supposed to make sure that servant does not construct the error responses, but lets `catchErrors` do that, but we moved away from that design.  now we're just testing that spar is not called with the wrong middleware stack.  we should make sure of that by writing two generic middleware stacks, one for servant, and one for wai-route.

So I removed the test after refactoring it a lot.  :-)